app/components/post_list/post_list.js

```diff
@@ -24,37 +24,74 @@ const style = StyleSheet.create({
     }
 });
 
-export default class PostList extends React.Component {
+const LOAD_MORE_POSTS = 'load-more-posts';
+
+export default class PostList extends Component {
     static propTypes = {
-        posts: React.PropTypes.array.isRequired,
-        theme: React.PropTypes.object.isRequired,
-        onPostPress: React.PropTypes.func,
-        renderReplies: React.PropTypes.bool,
-        indicateNewMessages: React.PropTypes.bool,
-        currentUserId: React.PropTypes.string,
-        lastViewedAt: React.PropTypes.number
+        posts: PropTypes.array.isRequired,
+        postsRequests: PropTypes.object.isRequired,
+        theme: PropTypes.object.isRequired,
+        allowLoadMore: PropTypes.bool,
+        loadMore: PropTypes.func,
+        onPostPress: PropTypes.func,
+        renderReplies: PropTypes.bool,
+        indicateNewMessages: PropTypes.bool,
+        currentUserId: PropTypes.string,
+        lastViewedAt: PropTypes.number
     };
 
     constructor(props) {
         super(props);
-        const {posts, indicateNewMessages, currentUserId, lastViewedAt} = this.props;
         this.state = {
+            hasFirstPost: false,
```

Comment:
Shouldn't `hasFirstPost` go in the store somewhere? Otherwise, we're going to have to keep checking this every time we switch into a channel